flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/EagerSchedulingStrategyTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.strategy;
+
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link EagerSchedulingStrategy}.
+ */
+public class EagerSchedulingStrategyTest extends TestLogger {
+
+	/**
+	 * Tests that when start scheduling eager scheduling strategy will start all vertices in scheduling topology.
+	 */
+	@Test
+	public void testStartScheduling() {
+		TestingSchedulingTopology testingSchedulingTopology = new TestingSchedulingTopology();
+		JobVertexID jobVertexID = new JobVertexID();
+		testingSchedulingTopology.addSchedulingVertex(new TestingSchedulingVertex(jobVertexID, 0));
+		testingSchedulingTopology.addSchedulingVertex(new TestingSchedulingVertex(jobVertexID, 1));
+		testingSchedulingTopology.addSchedulingVertex(new TestingSchedulingVertex(jobVertexID, 2));
+		testingSchedulingTopology.addSchedulingVertex(new TestingSchedulingVertex(jobVertexID, 3));
+		testingSchedulingTopology.addSchedulingVertex(new TestingSchedulingVertex(jobVertexID, 4));
+
+		TestingSchedulerOperation testingSchedulerOperation = new TestingSchedulerOperation();
+		EagerSchedulingStrategy schedulingStrategy = new EagerSchedulingStrategy(
+				testingSchedulerOperation,
+				testingSchedulingTopology,
+				null);
+
+		schedulingStrategy.startScheduling();
+
+		assertEquals(5, testingSchedulerOperation.getScheduledVertices().size());
```

Comment:
The assertions can be stricter. In particular, we should assert:

1. All vertices in the topology are scheduled at once, i.e., there is only one call to `allocateSlotsAndDeploy()` 
2. The deployed ExecutionVertexIDs match with the vertices in the topology.

To ease testing, I think it would help to make the constructor of `TestingSchedulingVertex` accept the `ExecutionVertexID` directly:
```
TestingSchedulingVertex(new ExecutionVetexID(...))
```

Using [hamcrest matchers](https://github.com/junit-team/junit4/wiki/matchers-and-assertthat):

```
assertThat(testingSchedulingOperations..., containsInAnyOrder(executionVertexId1, executionVertexId2))
```

Reply:
I refine the check to make it stricter. A little different with your proposal.